src/openrct2/actions/GameAction.cpp

```diff
@@ -445,6 +457,14 @@ namespace GameActions
             ActionLogContext_t logContext;
             LogActionBegin(logContext, action);
 
+            if (!(actionFlags & GameActions::Flags::ClientOnly))
+            {
+                auto undo = action->Get_Undo(*result);
+                if (undo != nullptr)
+                {
+                    _undoQueue.emplace_back(std::move(undo));
```

Comment:
perhaps move the placing in a queue until after completion of the Execute command.